src/shared/components/textIconArea/TextIconArea.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react';
+import { observer } from 'mobx-react';
+import classNames from 'classnames';
+import _ from 'lodash';
+import ReactDOM from 'react-dom';
+import { observable } from 'mobx';
+import autobind from "autobind-decorator";
+
+export interface ITextIconAreaProps {
+    elements: ITextIconAreaItemProps[];
+    text:string;
+    placeholder?: string;
+    classNames?: string[];
+    onIconClicked?: (itemId:string) => void;
```

Comment:
I left a comment on the previous commit, I'll add it here again because Im not sure how github will handle it/if it will be easy to see:

Could you change `itemId` to `itemValue` so that it's more clearly indicates what it is? It corresponds to `ITextIconAreaItemProps.value`, right?

Reply:
Done...